app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1173,8 +1173,12 @@ void displayMessage(String message, boolean bad, boolean suppressToast) {
     }
 
     public static boolean isDemoUser() {
-        User u = CommCareApplication._().getSession().getLoggedInUser();
-        return (User.TYPE_DEMO.equals(u.getUserType()));
+        try {
+            User u = CommCareApplication._().getSession().getLoggedInUser();
+            return (User.TYPE_DEMO.equals(u.getUserType()));
+        } catch (SessionUnavailableException e) {
+            return true;
```

Comment:
why does the session being unavailable mean that it's a demo user? Shouldn't we not be here at all if a `SessionUnavailableException` is thrown?